client.go

```diff
@@ -202,40 +104,65 @@ func (c *Client) writeTimeout() time.Duration {
 	return dnsTimeout
 }
 
-func (c *Client) exchange(ctx context.Context, m *Msg, a string) (r *Msg, rtt time.Duration, err error) {
-	var co *Conn
+func (c *Client) DialWithDialer(dialer *net.Dialer, address string) (conn *Conn, err error) {
+	// create a new dialer with the appropriate timeout
+	var realDialer net.Dialer
+	if dialer == nil {
+		realDialer = net.Dialer{}
+	} else {
+		realDialer = net.Dialer(*dialer)
+	}
+	realDialer.Timeout = c.getTimeoutForRequest(dialer, c.writeTimeout())
+
 	network := "udp"
-	tls := false
+	use_tls := false
```

Comment:
don't use underscores in Go names, also this does *not* have to be changed for this PR, so please bump to a new one.

Reply:
OK for renaming it.

I am not sure I get it right when you say this does not have to be changed for this PR. Do you mean "leave the variable name unchanged" or something else? In the former case, I changed it because the variable `tls` was clashing with the package name `tls`, and I found that `use_tls` explained better what it is used for

Reply:
done, diff incoming